status-im/status-desktop
# Speed up Admin Tab significantly

### What does the PR do

Iterates #16043
Is on top of #16279

Improves the speed of the Admin tab significantly by putting the different sections in Loaders, lazy loading them when needed.

It also fixes the members list speed a lot and also fixes the search.

It speeds up the permissions screen by using a ListView instead of a Repeater, that way, we don't instantiate all the items on start, only on scroll.

### Affected areas

Community admins panels and the Edit channel popup

### Architecture compliance

My PR is consistent with this document: [Architecture guidelines](../architecture-guide.md)

### Screenshot of functionality (including design for comparison)

:zap:  Video of how fast it is right now:

[fast-admin.webm](https://github.com/user-attachments/assets/27a57cd4-326a-4252-949a-78e411966997)

:snail: OLD version:

[slow-admin.webm](https://github.com/user-attachments/assets/6e97a6f5-4552-47ea-acf0-60b425a3a5a9)

As you can see, accessing the Admin panel is **way** faster now, since we lazy load. The only downside is that going to a particular tab, like the airdrop one, is a bit slower than before, because we load it when clicking on it, but a 0.5 second load is better than a 7+ seconds load on start.

You can also see, the members list search was excruciatingly slow. That is because it used the visibility of the elements instead of using the SortFilterProxyModel. It also didn't work well at all. It was case sensitive and only used the display name, so if you forgot that you put a nickname, it wouldn't work.

### Impact on end user

As seen in the video, the experience is way faster now and responsive.

### How to test

- Test all the admin panels and make sure they still navigate as expected (the content itself was not changed, just the loading, so in theory, no need to test the full airdrop for example)
- Test the edit channel popup to see if the permissions still show correctly

### Risk 


Tick **one**:

Worst case scenario is that one of the admin tabs is not accessible or the back buttons don't function. Or the permissions don't show correctly.
